src/lib/rnp.c

```diff
@@ -184,7 +184,10 @@ isarmoured(pgp_io_t *io, const char *f, const void *memory, const char *text)
     char       buf[BUFSIZ];
 
     armoured = 0;
-    (void) regcomp(&r, text, REG_EXTENDED);
+    if (regcomp(&r, text, REG_EXTENDED) != 0) {
+        fprintf(io->errs, "Can't compile regex\n");
+        return -1;
```

Comment:
why? `regcomp` can't compile the regex, so, I shoudn't try to free it. Am I wrong?